### ISSUE TITLE
Feature Request: make wagyu container neutral

### ISSUE THREAD

**Issue by xylosper** [mapbox/wagyu#103]
mapbox::geometry class templates are container neutral, that is, I can set container type as second template argument.

On the other hand, all geometry types in wagyu uses default argument which is std::vector.
It would be nice for wagyu to support arbitary container just like that mapbox::geometry does.

**Comment by flippmoke** [comment]
Yes this was something I really wanted to do in general, but I never got around to getting it done. This likely involve a large number of templating and a requirement for the user to create accessors methods of doing operations on their general geometry object.

I vaguely recall part of the reason I did this was to make it easier to copy out results and I used the geometry.hpp for internal calculations. I think all of this could easily be accomplished, but not planning right now to do this work due to personal limitations on time. I would happily support any PR on this.
